mm2src/coins/z_coin/z_coin_native_tests.rs

```diff
@@ -1,143 +1,155 @@
-use super::*;
-use crate::z_coin::z_htlc::z_send_dex_fee;
-use common::block_on;
-use common::now_ms;
+use bitcrypto::dhash160;
+use common::{block_on, now_ms};
 use mm2_core::mm_ctx::MmCtxBuilder;
+use mm2_test_helpers::for_tests::zombie_conf;
+use std::path::PathBuf;
 use std::time::Duration;
 use zcash_client_backend::encoding::decode_extended_spending_key;
 
+use super::{z_coin_from_conf_and_params_with_z_key, z_mainnet_constants, Future, PrivKeyBuildPolicy,
+            RefundPaymentArgs, SendPaymentArgs, SpendPaymentArgs, SwapOps, ValidateFeeArgs, ValidatePaymentError,
+            ZTransaction};
+use crate::z_coin::{z_htlc::z_send_dex_fee, ZcoinActivationParams, ZcoinRpcMode};
+use crate::{CoinProtocol, IguanaPrivKey};
+
 #[test]
 fn zombie_coin_send_and_refund_maker_payment() {
-    let conf = json!({
-        "coin": "ZOMBIE",
-        "asset": "ZOMBIE",
-        "fname": "ZOMBIE (TESTCOIN)",
-        "txversion": 4,
-        "overwintered": 1,
-        "mm2": 1,
-    });
-    let req = json!({
-        "method": "enable",
-        "coin": "ZOMBIE"
-    });
-
     let ctx = MmCtxBuilder::default().into_mm_arc();
-    let priv_key = [1; 32];
+    let mut conf = zombie_conf();
+    let params = default_zcoin_activation_params();
+    let priv_key = PrivKeyBuildPolicy::IguanaPrivKey(IguanaPrivKey::default());
```

Comment:
Please use `priv_key = [1; 32];` instead of default as it was done before, This will need changing in all tests that had it.
```diff
Index: mm2src/coins/z_coin/z_coin_native_tests.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mm2src/coins/z_coin/z_coin_native_tests.rs b/mm2src/coins/z_coin/z_coin_native_tests.rs
--- a/mm2src/coins/z_coin/z_coin_native_tests.rs	(revision 1d6c9afbaeed16f524db6d953f57b589297a5f2f)
+++ b/mm2src/coins/z_coin/z_coin_native_tests.rs	(date 1682094202752)
@@ -17,7 +17,8 @@
     let ctx = MmCtxBuilder::default().into_mm_arc();
     let mut conf = zombie_conf();
     let params = default_zcoin_activation_params();
-    let priv_key = PrivKeyBuildPolicy::IguanaPrivKey(IguanaPrivKey::default());
+    let private_key_data = [1; 32];
+    let priv_key = PrivKeyBuildPolicy::IguanaPrivKey(private_key_data.into());
     let db_dir = PathBuf::from("./for_tests");
     let z_key = decode_extended_spending_key(z_mainnet_constants::HRP_SAPLING_EXTENDED_SPENDING_KEY, "secret-extended-key-main1q0k2ga2cqqqqpq8m8j6yl0say83cagrqp53zqz54w38ezs8ly9ly5ptamqwfpq85u87w0df4k8t2lwyde3n9v0gcr69nu4ryv60t0kfcsvkr8h83skwqex2nf0vr32794fmzk89cpmjptzc22lgu5wfhhp8lgf3f5vn2l3sge0udvxnm95k6dtxj2jwlfyccnum7nz297ecyhmd5ph526pxndww0rqq0qly84l635mec0x4yedf95hzn6kcgq8yxts26k98j9g32kjc8y83fe").unwrap().unwrap();
     let CoinProtocol::ZHTLC(protocol_info) = serde_json::from_value::<CoinProtocol>(conf["protocol"].take()).unwrap() else { unimplemented!() };
@@ -27,7 +28,7 @@
         "ZOMBIE",
         &conf,
         &params,
-        priv_key.clone(),
+        priv_key,
         db_dir,
         z_key,
         protocol_info,
@@ -52,7 +53,6 @@
     };
     let tx = coin.send_maker_payment(args).wait().unwrap();
     println!("swap tx {}", hex::encode(tx.tx_hash().0));
-    let PrivKeyBuildPolicy::IguanaPrivKey(private_key_data) = priv_key else { unimplemented!() };
 
     let refund_args = RefundPaymentArgs {
         payment_tx: &tx.tx_hex(),
@@ -60,7 +60,7 @@
         other_pubkey: taker_pub,
         secret_hash: &secret_hash,
         swap_contract_address: &None,
-        swap_unique_data: private_key_data.as_slice(),
+        swap_unique_data: &private_key_data,
         watcher_reward: false,
     };
     let refund_tx = coin.send_maker_refunds_payment(refund_args).wait().unwrap();
```

Reply:
done

Reply:
As shown in the above patch, this line
https://github.com/KomodoPlatform/atomicDEX-API/blob/2340ad8c2d574f183cfb5f06444e19a6cde7a19b/mm2src/coins/z_coin/z_coin_native_tests.rs#L58
Won't be needed if we initialized private key data variable before using it in the policy. This will also avoid a clone that is currently used.

Reply:
This is not a blocker, I will approve the PR without it ,but it will make code a tiny bit better :)

Reply:
done